### PR TITLE
ENYO-5523: Update pointer mode after hiding VKB

### PR DESCRIPTION
**Pull request by germboy**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Spotlight's pointer mode isn't updated when attempting to switch modes while the VKB is visible (ex: when an input has focus).

### Resolution
While the VKB is visible, spotlight or currently-focused components (ex: `Input`) are unable to receive key events (`keydown`) events. During this time a user can change from pointer to 5-way mode. This prevents spotlight from restoring focus in some cases after dismissing the VKB, since spotlight falsely thinks it is still in pointer mode - which prevents programmatic focus changes.

Since we know pointer-mode can change while the VKB is open, we can remedy this issue by setting the platform pointer mode as a response to the webOS event `keyboardStateChange` when the VKB closes.

### CODE REVIEW

File: packages/spotlight/src/spotlight.js [hunk 1]
```diff
@@ -489,7 +495,10 @@ const Spotlight = (function () {
 				window.addEventListener('keyup', onKeyUp);
 				window.addEventListener('mouseover', onMouseOver);
 				window.addEventListener('mousemove', onMouseMove);
-				if (platform.webos) document.addEventListener('webOSMouse', handleWebOSMouseEvent);
+				if (platform.webos) {
+					document.addEventListener('webOSMouse', handleWebOSMouseEvent);
```
Comment (webOS101) [inline packages/spotlight/src/spotlight.js]: This will need to attach to the top-level document for use with iframes.

Comment (germboy) [reply packages/spotlight/src/spotlight.js]: of course :facepalm:

File: packages/spotlight/src/spotlight.js [hunk 2]
```diff
@@ -361,6 +361,12 @@ const Spotlight = (function () {
 		}
 	}
 
+	function handleKeyboardStateChangeEvent ({visibility}) {
```
Comment (teckliew) [inline packages/spotlight/src/spotlight.js]: Is `visibility` the state of VKB?

Comment (germboy) [reply packages/spotlight/src/spotlight.js]: Yes. When the state of the VKB changes (shows/hides), we receive a custom `keyboardStateChange` event that contains a `visibility` property that reflects the state of the VKB.

**Comment by ryanjduffy** [comment]
@teckliew Failed at first because of a CLI issue. Failed now because of a test failure surfaced because of the CLI update. Fortunately, it's fixed in `develop` so the push CI failure isn't meaningful.

**Pull request merged**